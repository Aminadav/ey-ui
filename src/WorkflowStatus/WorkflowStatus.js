import React from 'react';
import PropTypes from 'prop-types';
import Grid from '@material-ui/core/Grid';
import withStyles from '@material-ui/core/styles/withStyles';
import Typography from '@material-ui/core/Typography';
import ArrowLeftIcon from '@material-ui/icons/KeyboardArrowLeft';
import ArrowRightIcon from '@material-ui/icons/KeyboardArrowRight';
import WorkflowStatusItem from '../WorkflowStatusItem';

/**
 * Displays the current state of an editorial workflow
 *
 * @example ./__docs__/WorkflowStatus.md
 */
const WorkflowStatus = ({ status, label, classes }) => {
  return (
    <Grid item className={classes.grow}>
      <Grid container>
        {label && (
          <Typography
            data-testid="workflow-label"
            className={classes.label}
            variant="overline"
            gutterBottom
          >
            {label}
          </Typography>
        )}

        <WorkflowStatusItem title="Declined" active={status === 'declined'} status="declined" />
        <ArrowLeftIcon className={classes.arrow} />
        <WorkflowStatusItem title="Draft" active={status === 'draft'} status="draft" />
        <ArrowRightIcon className={classes.arrow} />
        <WorkflowStatusItem title="Pending" active={status === 'pending'} status="pending" />
        <ArrowRightIcon className={classes.arrow} />
        <WorkflowStatusItem title="Accepted" active={status === 'accepted'} status="accepted" />
      </Grid>
    </Grid>
  );
};

WorkflowStatus.propTypes = {
  status: PropTypes.string.isRequired,
  label: PropTypes.string,
  classes: PropTypes.shape({
    root: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
  }).isRequired,
};

WorkflowStatus.defaultProps = {
  label: null,
};

const styles = () => ({
  root: {
    display: 'flex',
  },
  label: {
    fontWeight: 'bold',
    padding: '0 1rem 0 0',
  },
  arrow: {
    margin: '.25rem 0',
    opacity: '.1',
  },
  grow: {
    flexGrow: 1,
  },
});

export default withStyles(styles)(WorkflowStatus);
