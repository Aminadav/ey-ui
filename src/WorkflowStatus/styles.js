import { makeStyles } from '@material-ui/styles';

const useStyles = makeStyles(() => ({
  root: {
    display: 'flex',
  },
  label: {
    fontWeight: 'bold',
    padding: '0 1rem 0 0',
  },
  arrow: {
    margin: '.25rem 0',
    opacity: '.1',
  },
  grow: {
    flexGrow: 1,
  },
}));

export default useStyles;
