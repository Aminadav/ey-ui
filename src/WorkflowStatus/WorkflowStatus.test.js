import React from 'react';
import { render } from 'react-testing-library';
import WorkflowStatus from './WorkflowStatus';

describe('WorkflowStatus', () => {
  it('matches the snapshot with label prop', () => {
    const { container } = render(<WorkflowStatus label="Agreement Status" status="draft" />);
    expect(container).toMatchSnapshot();
  });

  it('matches the snapshot without label prop', () => {
    const { container } = render(<WorkflowStatus status="draft" />);
    expect(container).toMatchSnapshot();
  });

  it('should render properly with a label element', () => {
    const { getByTestId } = render(<WorkflowStatus label="Agreement Status" status="draft" />);
    expect(getByTestId('workflow-label')).toHaveTextContent('Agreement Status');
  });

  it('should not render with a label element', () => {
    const { queryByTestId } = render(<WorkflowStatus status="draft" />);
    expect(queryByTestId('workflow-label')).toBeNull();
  });
});
