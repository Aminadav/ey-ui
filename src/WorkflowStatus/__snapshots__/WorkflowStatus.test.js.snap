// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkflowStatus matches the snapshot with label prop 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-grow-4 MuiGrid-item"
  >
    <div
      class="MuiGrid-root MuiGrid-container"
    >
      <span
        class="MuiTypography-root makeStyles-label-2 MuiTypography-overline MuiTypography-gutterBottom"
        data-testid="workflow-label"
      >
        Agreement Status
      </span>
      <span
        class="MuiTypography-root makeStyles-inactive-139 makeStyles-declinedStatus-140 MuiTypography-overline MuiTypography-gutterBottom"
      >
        Declined
      </span>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-arrow-3"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
        />
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
      </svg>
      <span
        class="MuiTypography-root makeStyles-active-138 makeStyles-draftStatus-141 MuiTypography-overline MuiTypography-gutterBottom"
      >
        Draft
      </span>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-arrow-3"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
        />
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
      </svg>
      <span
        class="MuiTypography-root makeStyles-inactive-139 makeStyles-pendingStatus-142 MuiTypography-overline MuiTypography-gutterBottom"
      >
        Pending
      </span>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-arrow-3"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
        />
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
      </svg>
      <span
        class="MuiTypography-root makeStyles-inactive-139 makeStyles-acceptedStatus-143 MuiTypography-overline MuiTypography-gutterBottom"
      >
        Accepted
      </span>
    </div>
  </div>
</div>
`;

exports[`WorkflowStatus matches the snapshot without label prop 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-grow-156 MuiGrid-item"
  >
    <div
      class="MuiGrid-root MuiGrid-container"
    >
      <span
        class="MuiTypography-root makeStyles-inactive-261 makeStyles-declinedStatus-262 MuiTypography-overline MuiTypography-gutterBottom"
      >
        Declined
      </span>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-arrow-155"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
        />
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
      </svg>
      <span
        class="MuiTypography-root makeStyles-active-260 makeStyles-draftStatus-263 MuiTypography-overline MuiTypography-gutterBottom"
      >
        Draft
      </span>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-arrow-155"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
        />
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
      </svg>
      <span
        class="MuiTypography-root makeStyles-inactive-261 makeStyles-pendingStatus-264 MuiTypography-overline MuiTypography-gutterBottom"
      >
        Pending
      </span>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root makeStyles-arrow-155"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
        />
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
      </svg>
      <span
        class="MuiTypography-root makeStyles-inactive-261 makeStyles-acceptedStatus-265 MuiTypography-overline MuiTypography-gutterBottom"
      >
        Accepted
      </span>
    </div>
  </div>
</div>
`;
